### DISCUSSION TITLE
Slow motion speed change

### DISCUSSION THREAD

**Discussion by AndreaMonzini** [p2or/blender-loom#64] [Ideas]
Would be possible to animate the speed change for the slow motion?

Example: 1.0, 1.1, 1.3, 1.6, 2.0, 2.5, 3.1, 3.8, 4.6, 5.5, 6.5 ...
To create a ramp for the speed.

Using the graph editor would be ideal like:
https://github.com/hxxr/time-remapper

Thank you.

**Comment by p2or** [comment]
Hi Andrea,

I really like the idea using the graph editor to draw a curve and use that information for rendering. Theoretically that's also easy to implement based on the existent *Render Selected Keyframes* operator and by resampling the curve.

However the problem is, even *No Auto Snap* is selected in the *Graph Editor* there is currently no way setting the timeline cursor to the actual frame as well as no way snapping the points to an even (rounded) number right? If so, I think aligning the curve properly can be time consuming and in especially frustrating at the end.

Any ideas how to deal with that issue? 

Cheers,
Christian

**Comment by AndreaMonzini** [comment]
Hello, i do not know if it can help but i see that there is an option in graph editor "Jump to Keyframe" "CRL+G" that correspond to `bpy.ops.graph.frame jump()`.
I just started to learn Python...

Here you can find more info about graph editor and time remapping by @Garrett-R:
https://blender.stackexchange.com/questions/1053/how-do-i-animate-time/7799#7799

where @ideasman42 suggested to use fcurve to animate time for `bpy.context.scene.frame_set()`

so @Garrett-R developed:
https://github.com/Garrett-R/time-remapper
and then it was forked more recently by @hxxr:
https://github.com/hxxr/time-remapper

Create the speed ramp directly with 3D rendering would be ideal, otherwise using the VSE or optical flow alternative like slowmoVideo ( great GPL software ) can create artifacts.

https://github.com/slowmoVideo/slowmoVideo
